### ISSUE TITLE
Arduino help sketch for iOS App

### ISSUE THREAD

**Issue by adamtelles** [DIT112-V21/group-08#29]
_The iOS app will generate a path and a set of instructions for the car to follow. These instructions will be communicated via MQTT to the car. The Arduino-MQTT sketch is a working sketch allowing the car to connect to a MQTT server. Advanced Obstacle Avoidance has a few good movement methods._

What will be needed in arduino for the iOS app to work is:
(In the iOS-App branch there is a sketch called '_arduinoMQTT_iOS.ino_' which already has a few of the functionalities. They may need some enhancements and testing to make sure that they work)

- Some sort of priority queue such that the car can follow the instructions in a specific order.
- A method that can take a distance as a parameter. Move the car that distance forward. And then stop.
- A method that can take some degrees as a parameter. Rotate the car that many degrees. And then stop.
- A method that can send speed / distance traveled via MQTT to the app.
- Some obstacle avoidance that make sure that if anything goes wrong, the car will at least not crash or get stuck.

**Comment by alex-nordin** [comment]
"A method that can take some degrees as a parameter. Rotate the car that many degrees. And then stop."
This method is in the arduino-tweaks branch. It is two methods, but it can be made into one pretty easily I think. It is usually off by about 3 degrees when turning, so if anyone has any idea why that is they can fix it.

**Comment by alex-nordin** [comment]
"A method that can take a distance as a parameter. Move the car that distance forward. And then stop."
Pushed this to the arduino-tweaks branch as well. Like the turning one it has some issues with accuracy, which is caused by the rate it updates. So to move the accurate distance the car has to move really slow. I don't know if this is a problem with my code or just an inevitable drawback of the sensors, but I won't be mad if someone tells me I just wrote dumb code. :)

"A method that can send speed / distance traveled via MQTT to the app."
I'll look at this one next.
